File: Graphs Gal Python/BellmanFord.py
import sys

INF = sys.maxsize            # infinito
N = 9                        # vertices
dist = [INF] * N             # vetor de distancias
pred = [-1] * N              # predecessores
adj = [[] for _ in range(N)] # vetor de adjascencia

def bellmanFord(s):
    # inicializando a distancia do vertice de partida como 0
    dist[s] = 0

    # atualizacao das distancias das arestas repetidamente (relaxamento)
    for i in range(N-1):
        alterado = False # flag que verifica o relaxamento
        for u in range(N):
            #se o vertice u ainda esta no menor caminho
            if dist[u] != INF:
                for v, w in adj[u]:
                    # relaxa a aresta
                    if dist[u] + w < dist[v]:
                        dist[v] = dist[u] + w
                        pred[v] = u
                        alterado = True

        # se nenhum vertice foi alterado, entao temos o fim do bellmanFord
        if not alterado:
            break

    # verificando a existencia de ciclos negativos
    for u in range(N):
        if dist[u] != INF:
            for v, w in dist[u]:
                if dist[u] + w < dist[v]:
                    return 1 # presenca de ciclo negativo
                
    return 0

File: Graphs Gal C++/BellmanFord.cpp
#include <bits/stdc++.h>

using namespace std;

#define INF 0x3f3f3f3f    // infinito
#define N 9// numero de vertices do grafo
vector<int> dist(N, INF); // vetor de distancias
vector<int> pred(N, -1);  // vetor de predecessores
vector<pair<int, int>> adj[N]; //vetor de adj
// prototipos de funcoes
int bellmanFord(int); // a funcao recebe o vertice de origem

int main()
{

}

int bellmanFord(int s)
{
    dist[s] = 0;
    for (int i = 0; i < N - 1; i++)
    {
        int alterado = 0;
        for (int u = 0; u < N; u++)
        {
            if (dist[u] != INF)
                for (auto p : adj[u])
                {
                    int v = p.first, w = p.second;
                    if (dist[u] + w >= dist[v])
                        continue;
                    dist[v] = dist[u] + w;
                    pred[v] = u;
                    alterado = 1;
                } // for auto
        }
        if (!alterado)
            break;
    } // for i

    // verifica ciclo negativo:
    // se possuir, retorna 1.
    // senao, retorna 0.
    for (int u = 0; u < N; u++)
        if (dist[u] != INF)
            for (auto p : adj[u])
            {
                int v = p.first, w = p.second;
                if (dist[v] > dist[u] + w)
                    return 1;
            } // for auto
    return 0;
} // belmannFord